effekt/shared/src/main/scala/effekt/lifted/LiftInference.scala

```diff
@@ -115,16 +116,21 @@ object LiftInference extends Phase[CoreTransformed, CoreLifted] {
 
       App(transform(b), targs, ev :: transformedArgs)
 
-    case core.Def(id, tpe, block, rest) =>
-      val env = pretransform(tree)
-      Def(id, tpe, transform(block)(using env, Context), transform(rest)(using env, Context))
+    case core.Scope(definitions, rest) =>
+      val env = pretransform(definitions)
+      val body = transform(rest)(using env, Context)
+
+      // TODO once we also port lifted to non-intrinsic lists, this can be simplified
+      definitions.foldRight(body) {
+        case (core.Definition.Def(id, tpe, block), rest) =>
+          Def(id, tpe, transform(block)(using env, Context), rest)
+        case (core.Definition.Let(id, tpe, binding), rest) =>
+          Let(id, tpe, transform(binding)(using env, Context), rest)
+      }
```

Comment:
Here we go back to the horrible intrinsic lists, to refactor one bit at a time. :)